docs/source/routing.rst

```diff
@@ -0,0 +1,240 @@
+                         :checked (done task)
+                         :onclick (make-js-action
+                                   (lambda (&key &allow-other-keys)
+                                     (toggle task))))
+                 (:span (if (done task)
+                            (with-html
+                              (:s (title task)))
+                            (:a :href (format nil "/tasks/~a" (id task)) ;; <-- only addition.
+                                (title task)))))))
+
+
+The task-page widget
+====================
+
+In Weblocks, an HTML block that we want to display, and possibly update
+independently, is a widget. Here, we want to show a task's details on
+their own page, it is then a widget.
+
+.. code-block:: common-lisp-repl
+
+   TODO> (defwidget task-page ()
+           ((task
+             :initarg :task
+             :initform nil
+             :accessor task-page-task)))
+
+   TODO> (defmethod render ((task-page task-page))
+           (let ((task (task-page-task task-page)))
+             (with-html
+               (:div "Task " (id task))
+               (:h1 (title task))
+               (:div (if (done task) "Done!" "To Do."))
+               (:div "Lorem ipsum…"))))
+
+
+Defining routes
+===============
+
+At this point we can think of our routes like this::
+
+.. code-block:: common-lisp
+
+    (defroutes tasks-routes
+      ("/tasks/\\d+" <create the task-page widget>)
+      ("/tasks/" (make-task-list)))
+
+The regexp ``\\d+`` will capture any url that is formed of digits and
+contains at least one.
+
+As we see, the ``task-page`` constructor will need to get the id
+matched by the route.
+
+
+Path and url parameters
+=======================
+
+To get the current path, use ``(weblocks/request:get-path)``. Then,
+you can find the matching parameters with ``cl-ppcre``.
+
+Our ``task-page`` constructor becomes:
+
+.. code-block:: common-lisp-repl
+
+   TODO> (defun make-task-page ()
+           (let* ((path (weblocks/request:get-path))
+                  (id (first (ppcre:all-matches-as-strings "\\d+" path)))
+                  (task (get-task (parse-integer id)))
+             (if task
+                 (make-instance 'task-page :task task)
+                 (not-found))))
+   TODO> (defun not-found ()
+           "Show a 404 not found page."
+           (with-html
+             (:div "Task not found.")))
+
+And our router is simply::
+
+   TODO> (defroutes tasks-routes
+           ("/tasks/task/\\d+" (make-task-page))
+           ("/tasks/" (make-task-list "Make my first Weblocks app"
+                                      "Deploy it somewhere"
+                                      "Have a profit")))
+
+The ``defroutes`` macro creates a new class and its constructor, named
+``make-<class-name>``.
+
+.. note:: It is important to use the constructor instead of ``make-instance``, as it defines properties on the fly.
+
+
+Final steps
+===========
+
+Make our router the main app for this session::
```

Comment:
@vindarel Just another minor detail. Should be widget instead of app: `Make our router the main widget for this session`.